### DISCUSSION TITLE
Appendix for the organization's 'readme'

### DISCUSSION THREAD

**Discussion by rgrupesh** [open-horizon/.github#7] [Ideas]
Adding appendix and glossary section will help new visitors to learn more about the project in short amount of time. What are the links and information that we should not miss? @joewxboy 

I would like to invite every community members to brainstorm their ideas below in the comment section.

**Comment by Akshit42-hue** [comment]
We can also add FAQ in the readme file that gives a clear idea for the new contributors
@rgrupesh . What's your take on this?

**Reply by rgrupesh** [reply]
That's good one. Please provide a list of questions that you think should be include in FAQs.

**Reply by joewxboy** [reply]
FAQ is currently on the documentation web site at https://open-horizon.github.io/docs/getting_started/faq.html
Is that the right place?  Should it live in the repo wiki?

**Reply by joewxboy** [reply]
Another potential item for the FAQ:

Q. I would like to start making open source contributions which I have never done before. I want to know what would be the expectation of the open source team with respect to making contributions each week to the project by a member.

A. These are typically volunteer efforts, and as such, any work performed should be based on how much time you have available.

Initially, it's a good idea to try to attend the short Working Group meetings which typically happen twice a month for 30-60 minutes, or alternatively watching the recording for any meeting you may have missed.  This will introduce you to the various persons working in this area within the project and alert you to the current activities and areas of concern.  Another good way to stay in touch is to monitor and participate in the chat rooms.  Last, if you sign up for the mailing list for the Working Group, you'll also get calendar invites to the meetings.

Some projects have larger tasks that may require a set number of hours to complete.  We usually reserve those tasks for persons dedicated to working on the project and/or mentees who have a dedicated number of hours to contribute.  For others, especially those just starting out, we typically recommend working on "good first issue" labelled items and then working your way up to larger issues that you're comfortable tackling.

**Reply by rgrupesh** [reply]
@joewxboy Since we already have FAQs at [https://open-horizon.github.io/docs/getting_started/faq.html](https://open-horizon.github.io/docs/getting_started/faq.html) and mentorship opportunities at [https://wiki.lfedge.org/display/OH/Mentorship+Programs](https://wiki.lfedge.org/display/OH/Mentorship+Programs), we could just provide link to that.  What other items you think is a must have in appendix section?

**Reply by Akshit42-hue** [reply]
Q. How to step up open-horizon in your local System
Q. Use of open-horizon
Q.What are the prerequisites for contributing in open-horizon

**Reply by joewxboy** [reply]
> Q. How to step up open-horizon in your local System

A. [Quick Start](https://open-horizon.github.io/quick-start), [How to install](https://open-horizon.github.io/common-requests/install.html), [All-in-One script and instructions](https://github.com/open-horizon/devops/tree/master/mgmt-hub), also for more advanced use cases - installing the [SmartAg Foundation developer stack](https://www.youtube.com/watch?v=mCqbzVYJyVs) (video)

> Q. Use of open-horizon

A. [How to use Open Horizon 101](https://open-horizon.github.io/common-requests/use.html), [Deploying a container](https://www.youtube.com/watch?v=Fk9zJyExELU&list=PLgohd895XSUddtseFy4HxCqTqqlYfW8Ix&index=3) (video), [Using Patterns and Policies](https://www.youtube.com/watch?v=vgUuOIefamA&list=PLgohd895XSUddtseFy4HxCqTqqlYfW8Ix&index=5) (video), [Model Management System Helper](https://www.youtube.com/watch?v=gWg_dphdGyA&list=PLgohd895XSUddtseFy4HxCqTqqlYfW8Ix&index=9&t=22s) (video), [Operator Helper](https://www.youtube.com/watch?v=2U4S8Epf418&list=PLgohd895XSUddtseFy4HxCqTqqlYfW8Ix&index=11) (video), [Using Secure Device Onboard](https://www.youtube.com/watch?v=PbJd7KX4_Ec&list=PLgohd895XSUddtseFy4HxCqTqqlYfW8Ix&index=12) (video)

> Q.What are the prerequisites for contributing in open-horizon

A. See the [Working on Issues](https://wiki.lfedge.org/display/OH/Project+Contributions) section on this wiki page

**Reply by joewxboy** [reply]
All of the above answers are only a start and could use a good Technical Writer's talent to summarize the information, place it in a logical order, and eventually to make lessons from it.  I [started on it](https://github.com/joe-diy/DIY-EdgeComputing/tree/master/CH01%20-%20Docker) two years ago, but got side-tracked, and now the information that has been written is already out-of-date.

**Reply by joewxboy** [reply]
@rgrupesh re: Appendix additions ... 

How to join the project, what roles exist and how to earn them

**Reply by rgrupesh** [reply]
> @rgrupesh re: Appendix additions ...
> 
> How to join the project, what roles exist and how to earn them

Noted.

**Comment by Ruchip16** [comment]
How about adding the mentorship opportunities they would get here refer [this](https://layer5.io/careers/programs)

**Reply by joewxboy** [reply]
We have a similar page here, but it's not as pretty as the Layer 5 page.

https://wiki.lfedge.org/display/OH/Mentorship+Programs

**Reply by joewxboy** [reply]
It may make sense to add it to the project web site, once we finish building it.

**Reply by joewxboy** [reply]
At the very least, it doesn't hurt to have it live in one place, and for all other places to link to it.

**Reply by Ruchip16** [reply]
Agreed, so at the moment we would just add existing links and on project completion, we would make the new webpage right?

**Comment by joewxboy** [comment]
@Rene-Ch1 Do we already have a Glossary in the Open Horizon documentation?

**Comment by Abhijay007** [comment]
@joewxboy we can also add a short description about each and every project that we have under open-horizon and provide a link to that particular repo, we can take reference from [here](https://github.com/layer5io)

**Reply by joewxboy** [reply]
Yes, but there are some repositories that can be combined into one entry.

Q: Would it make more sense to group the repositories together by logical component (Agent, Exchange, Switchboard, MSS, AgBot, Secrets Manager, SDO) and possibly provide an index somewhere else (at the page bottom) that lists repos alphabetically and maps to their logical component?  Or do you prefer to list all repos alphabetically and then have a section at the bottom that lists components and maps to their repos?  I think the former makes more sense, but I'd like all of your perspectives.

**Reply by Abhijay007** [reply]
@joewxboy yes I also think that grouping the repositories together by logical components is a better idea but instead of providing an index at the bottom, we can try to display it in the tabular form like [this](https://github.com/meshery/meshery#supported-service-meshes) as it is providing better navigation to user and we also sort them alphabetically too as it is kinda easy in this way.

**Comment by Ruchip16** [comment]
@joewxboy We can also create a GitHub repo for mentorships that Open-Horizon offers so that everyone can put their sample proposals there also further if Open Horizn takes part in GSOC or GSOD it would be much helpful at that time as well.

**Reply by joewxboy** [reply]
The concept is sound, but does a repo have too much overhead?  Would a wiki be more convenient?

**Reply by Ruchip16** [reply]
Ya wiki is also cool but already have one right?

**Reply by joewxboy** [reply]
We have both a Confluence-based wiki for the project community, and we have Github-based wikis for each repository inside each repo.

**Comment by Sukriti-sood** [comment]
Hey, It would be great if we will resize the open-horizon logo image, Currently it is covering more space, not good user exprience....?(Only to use in readme files)

**Reply by joewxboy** [reply]
Agreed, and if possible, have the logo align left and content on the right.

You absolutely do not want the logo to be so large that it takes up the whole screen. :)
